src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -544,6 +546,245 @@ def _get_connections(self):
         self._components_connection_outputs = openmdao_output_list
         self._components_connection_inputs = openmdao_input_list
 
+        if not self._check_existing_instance(self._power_train_file):
+            self._add_cache_instance(self._power_train_file)
+            self._check_connection(connections_list)
+            _LOGGER.info("Powertrain components' connections checked.")
+
+    def _check_connection(self, connections_list):
+        """
+        This function ensures that all the connections defined in the powertrain respect the
+        components connection limit. It also checks that no components are left unconnected and
+        ensure at least one propulsor is defined.
+
+        The _get_components method must be run beforehand.
+        """
+
+        # This should do nothing if it has already been run.
+        self._get_components()
+
+        propulsor_component = []
+        energy_storage_component = []
+        one_to_one_component = []
+        connector_component = []
+        connector_option = []
+        connector_type = []
+
+        for comp, option, type, comp_type in zip(
```

Comment:
I'd suggest something more explicit:
```suggestion
        for components_name, components_options, components_type_class, components_type in zip(
```